drafts/bitcartel-payment-disclosure/draft.rst

```diff
@@ -0,0 +1,345 @@
+
+When persisting, third party applications should expect serialization of payment disclosure to follow Zcash and upstream convention.
+
+::
+
+    ADD_SERIALIZE_METHODS;
+    ...
+    READWRITE(txid);
+    ...
+    READWRITE(joinSplitPrivKey);
+
+A new RPC call will be introduced to allow the sender to retrieve the payment disclosure data for a given shielded output index.
+
+``z_getpaymentdisclosure txid joinsplit_index output_index [message]``
+
+- Returns the payment disclosure in hexadecimal format.
+- Message is an optional parameter, a UTF-8 string.  We may want to restrict/sanitize this user input, e.g. number of characters, allowed characters.
+
+The sender wants the payment disclosure to be non-malleable, to prevent an attacker modifying details like the refund address.  To achieve this, the sender will sign the payment disclosure with the JoinSplitSig private key and append the signature to the end of the payment disclosure data.
+
+A new RPC call will be introduced to allow a third party to verify a payment disclosure.
+
+``z_validatepaymentdisclosure paymentdisclosure``
+
+Validates a payment disclosure and returns JSON output as follows:
+
+::
+
+    {
+      "txid": "68519fe52f2f64aa64e2a601e470fcde1069ab5a39652d277b7d816aa57169d1",
+      "jsIndex": 0,
+      "outputIndex": 0,
+      "version": 0,
+      "onetimeSymmetricKey": "09d6a2e2e8523280e5f56e79c080e82c4fe228086c92ea10986ca6d9bcea1d17",
+      "message": "howdy",
+      "joinSplitPubKey": "4c8d8135dca734b2c4fda25b2c7db29731d28a5f421f09da284b110bc0d1df91",
+      "signatureVerified": true,
+      "paymentAddress": "ztr4Ef2m7CFTtTvs4UpDz8zJY4Swukr3NRKThfLE12sNdGdav7Yf55G9HMAzGM3baR1FD43u9jb5JsAN67BBvz1UsVdLxoi",
+      "memo": "f600000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
+      "value": 10.00000000,
+      "commitmentMatch": true,
+      "valid": true
+    }
+
+Valid field is true if all the following conditions hold:
+
```

Comment:
- the payment disclosure is correctly encoded and has a recognized version

Reply:
ok